pkg/amqp/channelprovider.go

```diff
@@ -0,0 +1,222 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
```

Comment:
Hey @bstasyszyn, would you mind posting this PR without an additional license? TBH I'm not fully aware of the implications of that approach and I'd prefer to just follow the pattern used in all other repositories and not add any extra license inside.

Reply:
> Hey @bstasyszyn, would you mind posting this PR without an additional license? TBH I'm not fully aware of the implications of that approach and I'd prefer to just follow the pattern used in all other repositories and not add any extra license inside.

Will do. GoLand automatically adds this to the top new Go files and I didn't notice. I'll re-post.